_specifications/lsp/3.18/language/completion.md

```diff
@@ -525,6 +608,32 @@ export interface CompletionItemLabelDetails {
 }
 ```
 
+<div class="anchorHolder"><a href="#applyKind" name="applyKind" class="linkableAnchor"></a></div>
+
+```typescript
+/**
+ * Defines how values from a set of defaults and an individual item will be
+ * merged.
+ */
+export namespace ApplyKind {
+	/**
+	 * The value from the individual item (if provided and not `null`) will be
+	 * used instead of the default.
+	 */
+	export const Replace: 'replace' = 'replace;
+
+	/**
+	 * The value from the item will be merged with the default.
+	 * 
+	 * The specific rules for mergeing values are defined against each field
+	 * that supports merging.
```

Comment:
I added this because the two existing fields are handled differently (merging of two lists vs objects), but if we think that any mergable fields will always be a list that is merged in the same way (a union) or an object that is merged in the same way (a shallow merge), then we could move those two descriptions to here and remove them from the specific fields in `CompletionList.applyKind.x`. WDYT?

Reply:
I would leave them on the fields since it is easier to understand how it works.

Reply:
sgtm.

If everything else here looks good to you, I can start looking at the VS Code LSP client implementation - I want to be sure we're agreed on the spec first though :)

Thanks!